src/main/java/software/amazon/awssdk/crt/CRT.java

```diff
@@ -245,64 +245,23 @@ private static List<String> runProcess(String[] cmdArray) throws IOException {
         return output;
     }
 
-    private static void extractAndLoadLibrary(String path) {
-        try {
-            // Check java.io.tmpdir
-            String tmpdirPath;
-            File tmpdirFile;
-            try {
-                tmpdirFile = new File(path).getAbsoluteFile();
-                tmpdirPath = tmpdirFile.getAbsolutePath();
-                if (tmpdirFile.exists()) {
-                    if (!tmpdirFile.isDirectory()) {
-                        throw new IOException("not a directory: " + tmpdirPath);
-                    }
-                } else {
-                    tmpdirFile.mkdirs();
-                }
 
-                if (!tmpdirFile.canRead() || !tmpdirFile.canWrite()) {
-                    throw new IOException("access denied: " + tmpdirPath);
-                }
-            } catch (Exception ex) {
-                String msg = "Invalid directory: " + path;
-                throw new IOException(msg, ex);
-            }
-
-            String libraryName = System.mapLibraryName(CRT_LIB_NAME);
-
-            // Prefix the lib we'll extract to disk
-            String tempSharedLibPrefix = "AWSCRT_";
-
-            File tempSharedLib = File.createTempFile(tempSharedLibPrefix, libraryName, tmpdirFile);
-            if (!tempSharedLib.setExecutable(true, true)) {
+    /**
+     * Extract the CRT JNI library on current platform to a specific FIle.
+     */
+    public static void extractLibrary(File extractFile) {
```

Comment:
Can we have a note either here or in readme on how the users are supposed to use it and what the pitfalls are.